probe-rs/src/debug/variable.rs

```diff
@@ -372,49 +372,41 @@ impl Variable {
         memory: &mut impl MemoryInterface,
         variable_cache: &mut variable_cache::VariableCache,
         new_value: String,
-    ) -> Result<String, DebugError> {
-        let variable_name = if let VariableName::Named(variable_name) = &self.name {
-            variable_name.clone()
+    ) -> Result<(), DebugError> {
+        let is_pointer = if let VariableName::Named(variable_name) = &self.name {
+            variable_name.starts_with('*')
         } else {
-            String::new()
+            false
         };
-        let updated_value = if !self.is_valid()
+
+        if !self.is_valid()
                 // Need a valid type
                 || self.type_name == VariableType::Unknown
                 // Need a valid memory location
                 || !self.memory_location.valid()
         {
             // Insufficient data available.
-            return Err(anyhow!(
+            Err(anyhow!(
                 "Cannot update variable: {:?}, with supplied information (value={:?}, type={:?}, memory location={:#010x?}).",
-                self.name, self.value, self.type_name, self.memory_location).into());
-        } else if variable_name.starts_with('*') {
+                self.name, self.value, self.type_name, self.memory_location).into())
+        } else if is_pointer {
```

Comment:
Why are pointers special-cased here? They should just fail in `update_variable` anyway, shouldn't they.

Reply:
This is just code which was there before. I agree that this might be unnecessary, I just wanted to clean it up at bit.